apps/src/dance/ai/DanceAiClient.js

```diff
@@ -10,93 +10,77 @@ import CachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foregroun
  * example: {"backgroundEffect":"sparkles","backgroundColor":"cool","foregroundEffect":"bubbles"}
  */
 export function chooseEffects(emojis) {
-  var foregroundValues = {};
-  var backgroundValues = {};
-  var paletteValues = {};
-  reformatInputs(foregroundValues, CachedForegrounds);
-  reformatInputs(backgroundValues, CachedBackgrounds);
-  reformatInputs(paletteValues, CachedPalettes);
+  const outputTypes = [CachedBackgrounds, CachedForegrounds, CachedPalettes];
+  const outputWeights = [];
+  outputTypes.forEach(outputJson => {
+    outputWeights.push(calculateOutputWeightsVector(emojis, outputJson));
+  });
```

Comment:
nit: can be `const outputWeights = outputTypes.map(outputJson => calculateOutputWeightsVector(emojis, outputJson));`